app/lua/lobject.h

```diff
@@ -47,6 +47,18 @@ typedef union GCObject GCObject;
 */
 #define CommonHeader	GCObject *next; lu_byte tt; lu_byte marked
 
+#ifdef __XTENSA__
+/* force assess to tt and marked fields through 32-access */
+/* tt = .... B...   marked = .... .B.. */
+#define GET_BYTE_FN(name,wo,bo) \
```

Comment:
I am not sure how we could this arithmetic e.g. to convert `__builtin_offsetof()` for `tt` and `marked` (4 and 5 resp) into 1+0 and 1+8 for feeding into the asm() primitive which only takes string arguments.  So string concat as in ``"l32i  %0, %1, " #wo "; extui %0, %0, " #bo ", 8;"` works but compile-time arithmetic in these expressions doesn't, so I couldn't use the extui instruction and would need to add extra instructions (4, I think) to do this at runtime.

Note sure this is worth it.

Reply:
1? You do realise the value you've used here is 4, right? :)

I had thought that simply taking the wo/bo as arguments (rather than token pasting) would have made it work. I could be wrong, feel free to disregard. In that case, maybe it would be worthwhile leaving an "assert bomb" in the form of
```
static const char the_whatsimajigit_struct_has_changed_format[__builtin_offsetof(yourstruct, tt) == 4 ? 0 : -1];
```
to draw attention to the problem at compile time, if it were to occur (please substitute correct names in the above).